cmd/status.go

```diff
@@ -0,0 +1,192 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/elastic-package/internal/cobraext"
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/packages/changelog"
+	"github.com/elastic/elastic-package/internal/packages/status"
+	"github.com/elastic/elastic-package/internal/registry"
+)
+
+const statusLongDescription = `Use this command to display the current deployment status of a package.
+
+If a package name is specified, then information about that package is
+returned, otherwise this command checks if the current directory is a
+package directory and reports its status.`
+
+func setupStatusCommand() *cobraext.Command {
+	cmd := &cobra.Command{
+		Use:   "status [package]",
+		Short: "Show package status",
+		Args:  cobra.MaximumNArgs(1),
+		Long:  statusLongDescription,
+		RunE:  statusCommandAction,
+	}
+	cmd.Flags().BoolP(cobraext.ShowAllFlagName, "a", false, cobraext.ShowAllFlagDescription)
+
+	return cobraext.NewCommand(cmd, cobraext.ContextPackage)
+}
+
+func statusCommandAction(cmd *cobra.Command, args []string) error {
+	var packageName string
+
+	showAll, err := cmd.Flags().GetBool(cobraext.ShowAllFlagName)
+	if err != nil {
+		return cobraext.FlagParsingError(err, cobraext.ShowAllFlagName)
+	}
+	if len(args) > 0 {
+		packageName = args[0]
+	}
+	packageStatus, err := getPackageStatus(packageName, showAll)
+	if err != nil {
+		return err
+	}
+	return print(packageStatus, os.Stdout)
+}
+
+func getPackageStatus(packageName string, showAll bool) (*status.PackageStatus, error) {
+	if packageName != "" {
+		return status.RemotePackage(packageName, registry.SearchOptions{
+			All:          showAll,
```

Comment:
Actually I was thinking to expose all these options with CLI and define default variables there.

Reply:
so, when would you ever not want to set `experimental` or `internal`?

Reply:
So, resolution we chatted about off github was to keep `experimental`/`internal` set to true for this particular command. In the future, adding some sort of `list` or `search` command would warrant making these configurable.